### ISSUE TITLE
Lost whitespace in power point

### ISSUE THREAD

**Issue by lklein53** [OfficeDev/Open-XML-SDK#848]
Hi,
I encountered the following issue using the OpenXML SDK

**Description**
I am reading in a simple pptx file (see the resource in the repro section) using the OpenXML SDK (code provided in the repro section).
First opening the file and then reading the slide part with the OpenXMLReader.
The text element containing only white spaces is trimmed and after reading contains no characters.
This behaviour destroys the layout of manually indented content if the file would be saved after reading it.

Similar reports:
https://stackoverflow.com/questions/6106044/missing-whitespace-within-pptx-paragraphs-using-openxmlsdk?rq=1
https://stackoverflow.com/questions/7153387/powerpoint-openxml-whitespace-is-disappearing?rq=1

**Information**

- .NET Target: .NET Core
- DocumentFormat.OpenXml Version: 2.11.3

**Repro**
I have created a small example using an example pptx file with one slide
https://github.com/lklein53/openxmlsdk-whitespace

**Observed**

The xml reader used internally is configured to ignore insignificant whitespaces which results in the removal of the whitespaces in the text element. I haven't seen the possibility to change that behaviour. For word documents there is the xml:space=preserve attribute added to such text nodes which seems to be not the case for power point files. 

**Expected**

It would be nice if it would be possible to configure the current behaviour to explicitly keep insignificant whitespaces if that is desired by the user of the SDK. I could implement this functionality if you think it should be added.

Please let me know if I am using the OpenXMLReader wrong and there is another way to read in the slide part.

Thanks for your help and kind regards,
Lars

(edit by @twsouthwick)

```csharp
using System;
using System.Linq;
using System.Reflection;
using DocumentFormat.OpenXml;
using DocumentFormat.OpenXml.Drawing;
using DocumentFormat.OpenXml.Packaging;
using Xunit;
using Path = System.IO.Path;

namespace TestProject1 {
public class UnitTest1 {
  [Fact]
  public void Test1() {
    const string fileName = "Resources/example.pptx";
    var path = Path.Combine(Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location) ?? throw new InvalidOperationException(), fileName);
    using var doc = PresentationDocument.Open(path, false);
    var slide = doc.PresentationPart.SlideParts.First();
    var reader = OpenXmlReader.Create(slide);
    reader.Read();
    var element = reader.LoadCurrentElement();
    var text = element.Descendants<Text>().First();
    Assert.Equal(" ", text.Text);
  }
}
}
```

**Comment by adamshakhabov** [comment]
Interesting, I see that XML content also doesn't contain any whitespace😕:

![image](https://user-images.githubusercontent.com/5969304/104228320-5630f580-5453-11eb-9623-1792baf23c55.png)

Your text shape contains _tab_, not _whitespace_, by the way.

**Comment by lklein53** [comment]
Hi, thanks for checking the example. Yes you are right the tab is a mistake in my example file. I have changed it to a whitespace so the description of the issue matches the reporducer.

**Comment by adamshakhabov** [comment]
Check it:

```C#
using P = DocumentFormat.OpenXml.Presentation;
using A = DocumentFormat.OpenXml.Drawing;

namespace SDKTesting
{
    class Program
    {
        static void Main(string[] args)
        {
            PresentationDocument presentationDoc = PresentationDocument.Open(@"c:\example.pptx", false);
            P.ShapeTree shapeTree = presentationDoc.PresentationPart.SlideParts.First().Slide.CommonSlideData.ShapeTree;
            A.Text aText = shapeTree.Descendants<A.Text>().First();
            
            Console.WriteLine($"Hello{aText.Text}World!");
        }
    }
}
```

> Hello World!

**Comment by lklein53** [comment]
Hi,
Thanks for providing the working example. So it seems having a generic interface that just takes a OpenXmlPart and return an OpenXmlElement is not possible without loosing whitespaces in pptx ?

```csharp
  public OpenXmlElement readPart(OpenXmlPart part) {
    var reader = OpenXmlReader.Create(part);
    reader.Read();
    return reader.LoadCurrentElement();
  }
```

**Comment by twsouthwick** [comment]
Hmmm this is interesting. Note that the example @adamshakhabov works, but the code it uses internally is not just creating a part and loading it. I'd suggest debugging into that to see what the difference is (the SDK supports source link (see https://docs.microsoft.com/en-us/dotnet/standard/library-guidance/sourcelink for details) so you can just step into its methods

**Comment by lklein53** [comment]
The working code from @adamshakhabov uses the `LoadFromPart` method of the OpenXmlPartRootElement which uses 
`XmlConvertingReaderFactory.Create(partStream, OpenXmlElementContext.XmlReaderSettings, openXmlPart.OpenXmlPackage.StrictRelationshipFound))` with different XmlReaderSettings than the OpenXmlReader. 

The XmlReaderSettings of the OpenXmlReader explicitly set `IgnoreWhitespace = true` where the OpenXmlElementContext explicitly sets it to false
 // Set IgnoreWhitespace to false for the SDK to handle the whitespace node type. We have to do this because
// PPT does not use the preserve attribute (xml:space="preserve") for non-ignorable whitespace.
IgnoreWhitespace = false

**Comment by twsouthwick** [comment]
Ah so are you saying you'd like to add a parameter to OpenXmlReader that accepts a setting option? I don't see anything wrong with that at this time. Happy to take a PR with that.

**Comment by lklein53** [comment]
Hi,
Thanks for your reply. I will create a PR with the changes

**Comment by twsouthwick** [comment]
Fixed in #857. Thanks!
